src/memory/shop_manager.rs

```diff
@@ -0,0 +1,74 @@
+use crate::memory::memory_context::MemoryContext;
+use crate::memory::{MemoryManager, MemoryManagerUpdate};
+use crate::state::StateContext;
+use log::info;
+use memory::game_engine::il2cpp::unity_list::{UnityItem, UnityList};
+use memory::memory_manager::il2cpp::UnityMemoryManager;
+use memory::process::MemoryError;
```

Comment:
I'll knock out the ones that make sense:
i usually like to keep things on one scope together, because once you start collating too much, cargo fmt makes a huge mess out of them